parlai/core/torch_agent.py

```diff
@@ -1660,14 +1660,26 @@ def observe(self, observation):
             # make sure we note that we're expecting a reply in the future
             self.__expecting_to_reply = True
 
+        # keep around the observation for updating history based on label
         self.observation = observation
+
+        # possibly change tokenization methodology based on if this is a
+        # training example
+        if hasattr(self.dict, 'set_training_mode'):
```

Comment:
I'm lightly confused by this piece of code. `set_training_mode` is a function in DictAgent but this `if` seems to be looking for it as a member variable? 

(Noticing that this is stacked on top of another diff so maybe it's written there, but makes it a tad tricky to review if so.)

Reply:
Functions are also attributes

Reply:
Are there cases where `self.dict` isn't `DictionaryAgent`? Cause otherwise, this will always return `True`, no?